lms/djangoapps/instructor/services.py

```diff
@@ -78,13 +76,25 @@ def delete_student_attempt(self, student_identifier, course_id, content_id, requ
             except (StudentModule.DoesNotExist, enrollment.sub_api.SubmissionError):
                 err_msg = (
                     'Error occurred while attempting to reset student attempts for user '
-                    '{student_identifier} for content_id {content_id}.'.format(
-                        student_identifier=student_identifier,
-                        content_id=content_id
-                    )
+                    f'{student_identifier} for content_id {content_id}.'
                 )
                 log.error(err_msg)
 
+    def complete_student_attempt(self, user_identifier, content_id):
+        """
+        Calls the complete_student_attempt_task which
+        Submits all completable xblocks inside of the content_id block to the
+        Completion Service to mark them as complete. One use case of this function is
+        for special exams (timed/proctored) where regardless of submission status on
+        individual problems, we want to mark the entire exam as complete when the exam
+        is finished.
+
+        params:
+            user_identifier (String): username or email of a user
+            content_id (String): the block key for a piece of content
```

Comment:
I think normally we just use `str` to mark strings.

Also! Did you know that python3 has fancy annotations for arguments / return values? Like you can write this:

```
     def complete_student_attempt(self, user_identifier: str, content_id: str) -> None: 
```

Not something that you need to / should do here. We barely have any annotations around the place. But a cool thing that I didn't know much about until recently.

https://www.python.org/dev/peps/pep-3107/